src/bindgen/ir/function.rs

```diff
@@ -64,10 +65,24 @@ impl Function {
             ret.replace_self_with(self_path);
         }
 
+        // Check for associated type in ret
+        let mut ret_assoc = None;
+        if let syn::ReturnType::Type(_, ret_type) = &sig.output{
+            if let syn::Type::Path(type_path) = ret_type.as_ref(){
+                if let Some(_) = type_path.qself{ 
+                    let assoc_exist = AssocTypeId::new(&type_path);
+                    match assoc_exist{
+                        Ok(id) => ret_assoc = Some(id),
+                        Err(msg) => warn!("Err parsing assoc-type: {}", msg),
+                    }
+                }
+            }
```

Comment:
```suggestion
            if let syn::Type::Path(type_path) = &*ret_type {
                ret_assoc = type_path.qself.map(|_| AssocTypeId::new(&type_path)).transpose()?;
            }
```

or did you write it like that on purpose? In the sense that you didn't want the parsing to fail completely